CONTRIBUTING.md

```diff
@@ -155,6 +164,85 @@ git push origin HEAD # This will close the PR
+All pages and components should be written using the Composition API, and not the standard Options API.
+If you are unfamiliar with the Composition API, you can read more about it [here](https://v3.vuejs.org/guide/composition-api-introduction.html#why-composition-api).
+
+Check out [`GrantRegistryNewGrant.vue`](app/src/views/GrantRegistryNewGrant.vue) for an example of a component that uses the Composition API.
+In the Composition API docs referenced earlier, most examples put everything within the `setup()` method.
+This is ok for simple components. For more complex components, logic and variables should be moved in composition functions, like the `useNewGrant()` method defined in `GrantRegistryNewGrant.vue`.
+
+### State Management
+
+This app uses its own state management pattern based on the Composition API, largely based on the pattern detailed in [this article](https://vueschool.io/articles/vuejs-tutorials/state-management-with-composition-api/).
+The comments in [`wallet.ts`](app/src/store/wallet.ts) also provide details on this store pattern.
+
+As a general rule of thumb, whenever you need to read or fetch more data from somewhere, it should be done in:
+- `wallet.ts` for anything related to the user's wallet, e.g. token balances or ENS name
+- `data.ts` for generic data fetching, such as reading a list of grant data from the `GrantRegistry` contract
+- `cart.ts` for anything related to a user's cart and the checkout process
+- `settings.ts` for user settings, which are saved in the browser's local storage
+
+If you are writing a component or page that derives something from data that's already in a store, use a computed property and export the data from the store.
+See the `userDisplayName` example in `wallet.ts` as an example.
+That computed property is inlined since it's simple, but more complex computed properties can be extracted like `cartSummary` in `cart.ts`. If your computed property needs to be async, use a watcher instead.
+
+Not every state variable, method, and computed property needs to be exported from the store module.
+Only export aspects that are needed in other components or pages to keep interfaces and intentions clean and clear.
+
+Do not export state variables directly from the store, as this is almost never required.
+This is because in general state should not be mutated directly, but instead should only be mutated through a method exported from the store. In `cart.ts` we directly export the `cart` variable for simplicity when a user is editing their cart. However, a cleaner way might be to create a local copy of `cart` when `Cart.vue` mounts, and have `Cart.vue` call `updateCart` whenever the user edits their cart.
+
+### Code Style Guide
+
+In general, the app follows the recommendations of the official [Vue Style Guide](https://vuejs.org/v2/style-guide/), which should be followed as much as possible.
```

Comment:
Style Guide V3 link: https://v3.vuejs.org/style-guide/

Reply:
Ah nice, done in 5a5d69d961814643f76f99a56755449d62564325